workflows/flows/import_genome_assembly_links.py

```diff
@@ -0,0 +1,303 @@
+
+# Import Django models
+from analyses.models import Assembly
+from genomes.models import Genome, GenomeAssemblyLink
+from workflows.data_io_utils.csv.csv_comment_handler import (
+    CommentAwareDictReader,
+    CSVDelimiter,
+)
+
+
+@task(name="Validate TSV file")
+def validate_tsv_file(tsv_path: str) -> str:
+    """
+    Validates that the TSV file exists and is readable.
+
+    Args:
+        tsv_path: Path to the TSV file
+
+    Returns:
+        The validated path to the TSV file
+
+    Raises:
+        FileNotFoundError: If the TSV file does not exist
+    """
+    logger = get_run_logger()
+    path = Path(tsv_path)
+
+    if not path.exists():
+        logger.error(f"TSV file not found: {tsv_path}")
+        raise FileNotFoundError(f"TSV file not found: {tsv_path}")
+
+    if not path.is_file():
+        logger.error(f"Path is not a file: {tsv_path}")
+        raise ValueError(f"Path is not a file: {tsv_path}")
+
+    logger.info(f"TSV file validated: {tsv_path}")
+    return tsv_path
+
+
+@task(name="Read TSV file")
+def read_tsv_file(tsv_path: str) -> List[Dict[str, str]]:
+    """
+    Reads the TSV file and returns a list of dictionaries, one for each row.
+
+    Args:
+        tsv_path: Path to the TSV file
+
+    Returns:
+        List of dictionaries, one for each row in the TSV file
+    """
+    logger = get_run_logger()
+    records = []
+
+    with open(tsv_path, "r") as f:
+        reader = CommentAwareDictReader(
+            f, delimiter=CSVDelimiter.TAB, none_values=["", "NA", "N/A", "null", "NULL"]
+        )
+
+        # Check if required columns are present
+        required_columns = ["primary_assembly", "genome"]
+        missing_columns = [
+            col for col in required_columns if col not in reader.fieldnames
+        ]
+
+        if missing_columns:
+            logger.error(
+                f"Missing required columns in TSV: {', '.join(missing_columns)}"
+            )
+            raise ValueError(
+                f"Missing required columns in TSV: {', '.join(missing_columns)}"
+            )
+
+        for row in reader:
+            records.append(row)
+
+    logger.info(f"Read {len(records)} records from TSV file")
+    return records
+
+
+@task(name="Process TSV records")
+def process_tsv_records(records: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    """
+    Processes the TSV records, validating and cleaning the data.
+
+    Args:
+        records: List of dictionaries, one for each row in the TSV file
+
+    Returns:
+        List of validated and cleaned records
+    """
+    logger = get_run_logger()
+    validated_records = []
+
+    for i, record in enumerate(records, 1):
+        # Check for required fields
+        if not record.get("primary_assembly"):
+            logger.warning(f"Row {i}: Missing primary_assembly, skipping")
+            continue
+
+        if not record.get("genome"):
+            logger.warning(f"Row {i}: Missing genome, skipping")
+            continue
```

Comment:
I think you can skip these two as it was already validated by `read_tsv_file`... if I'm reading this correctly